message.go

```diff
@@ -743,6 +753,20 @@ func (m *RedisMessage) AsFloatSlice() ([]float64, error) {
 	return s, nil
 }
 
+// AsBoolSlice checks if the message is a redis array/set response, and converts it to []bool.
+// Redis nil elements and other non-boolean elements will be present as false.
+func (m *RedisMessage) AsBoolSlice() ([]bool, error) {
+	values, err := m.ToArray()
+	if err != nil {
+		return nil, err
+	}
+	s := make([]bool, len(values))
+	for i, v := range values {
+		s[i] = v.string == "1"
```

Comment:
I think your original use of `strconv.ParseBool` was good. Why did you remove it completely? I believe we better add it to AsBool() and re-use AsBool() here.